File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> ans;
        int i = 0; 
        int num = 1;
        
        while (i < target.size() && num <= n) {
            if (target[i] == num) {
                ans.push_back("Push");
                i++;
            } else {
                ans.push_back("Push");
                ans.push_back("Pop");
            }
            num++;
        }
        
        return ans;
    }
};

File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.go
func buildArray(target []int, n int) []string {
    ans := []string{}
    i := 0   
    num := 1  
    
    for i < len(target) && num <= n {
        if target[i] == num {
            ans = append(ans, "Push")
            i++
        } else {
            ans = append(ans, "Push", "Pop")
        }
        num++
    }
    
    return ans
}